### ISSUE TITLE
Broken layout for Badge with 'variant=dot' and right section

### ISSUE THREAD

**Issue by Toshinaki** [mantinedev/mantine#6751]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

7.2.12

### What package has an issue?

@mantine/core

### What framework do you use?

Vite

### In which browsers you can reproduce the issue?

Chrome

### Describe the bug

`Badge` with `variant='dot'` layout broken when using `rightSection={xxx}`
please check the codesandbox

### If possible, include a link to a codesandbox with a minimal reproduction

https://codesandbox.io/p/sandbox/gifted-http-n33kty?file=%2Fsrc%2Findex.tsx

### Possible fix

maybe this css:
`grid-template-columns: auto 1fr auto;`

for `[data-variant='dot']`, it's `grid-template-columns: auto 1fr;`

### Self-service

- [ ] I would be willing to implement a fix for this issue

**Comment by Toshinaki** [comment]
I'm currently using this css rule for workaround:
```
.mantine-Badge-root:where(
  [data-with-left-section],
  [data-with-right-section],
  [data-variant='dot']
) {
  grid-template-columns: auto 1fr auto;
}
```

**Comment by pratyush369** [comment]
I can work on fixing it

**Comment by rtivital** [comment]
Fixed in [7.13.2](https://github.com/mantinedev/mantine/releases/tag/7.13.2)
